server-spi/src/main/java/org/keycloak/models/UserSessionProvider.java

```diff
@@ -208,4 +209,7 @@ UserSessionModel createUserSession(String id, RealmModel realm, UserModel user,
     void close();
 
     int getStartupTime(RealmModel realm);
+
+    default void migrate(String modelVersion) {
```

Comment:
We can change the declaration to `migrate(ModelVersion version)` and then we don't need to create a new `ModelVersion` instance in `PersistentUserSessionProvider`.

Reply:
Unfortunately this is not possibly now as the ModelVersion is not a public API. So I fell back to a String. IMHO this needs to be resolved outside of this PR.

Reply:
Agreed. I didn't notice the `*-private` suffix on the module.